src/core/cleanup.ts

```diff
@@ -16,4 +16,13 @@ function removeCleanup(callback: () => Promise<void> | void) {
   cleanupCallbacks = cleanupCallbacks.filter((cb) => cb !== callback)
 }
 
+cleanup.autoRegister = () => {
```

Comment:
I'm second guessing myself on whether it's a good idea to put this on `cleanup` or to just have it as another export.  I think currently it will be exposed out of the public API, which might be ok, but then we should to document and maintain it.

Reply:
Don't change this yet... I'll sleep on it.

Reply:
Still undecided. I'm leaning towards just another export.